src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -48,32 +46,121 @@ final case class EServicesApiServiceImpl(
+      requesterTenant <- tenantManagementService
+        .getTenant(eService.producerId)
+      agreement       <- agreementProcessService
+        .getAgreements(
+          consumerId = requesterId.some,
+          eServiceId = eserviceId.some,
+          descriptorId = descriptorId.some,
+          states = Seq.empty,
+          latest = None
+        )
+        .map(_.headOption)
+    } yield CatalogEServiceDescriptor(
+      eServiceId = eService.id,
+      name = eService.name,
+      eServiceDescription = eService.description,
+      technology = eService.technology.toApi,
+      attributes = convertToApiAttributes(eService.attributes, attributes.attributes),
+      descriptorId = descriptor.id,
+      version = descriptor.version,
+      descriptorDescription = descriptor.description,
+      interface = descriptor.interface.map(_.toApi),
+      docs = descriptor.docs.map(_.toApi),
+      state = descriptor.state.toApi,
+      audience = descriptor.audience,
+      voucherLifespan = descriptor.voucherLifespan,
+      dailyCallsPerConsumer = descriptor.dailyCallsPerConsumer,
+      dailyCallsTotal = descriptor.dailyCallsTotal,
+      agreementApprovalPolicy = descriptor.agreementApprovalPolicy.toApi,
+      activeDescriptor = getActiveDescriptor(eService).map(ad => CompactDescriptor(ad.id, ad.state.toApi, ad.version)),
+      agreement = agreement.map(a => CompactAgreement(id = a.id, state = a.state.toApi)),
+      canSubscribe = certifiedAttributesSatisfied(
+        eService.attributes.toManagement,
+        requesterTenant.attributes.mapFilter(_.certified)
+      ),
+      isSubscribed = agreement.exists(a => SUBSCRIBED_AGREEMENT_STATES.contains(a.state)),
+      isMine = eService.producerId.toString == requesterId,
+      mail = requesterTenant.mails.headOption.map(_.toApi)
+    )
 
     onComplete(result) {
       handleError(s"Error retrieving descriptor $descriptorId of eservice $eserviceId") orElse {
         case Failure(x: EServiceDescriptorNotFound) =>
           logger.warn(x.getMessage)
-          getEserviceDescriptor404(problemOf(StatusCodes.NotFound, x))
-        case Success(descriptor)                    => getEserviceDescriptor200(descriptor)
+          getCatalogEServiceDescriptor404(problemOf(StatusCodes.NotFound, x))
+        case Success(descriptor)                    => getCatalogEServiceDescriptor200(descriptor)
       }
     }
   }
 
+  private def extractIdsFromAttributes(attributes: CatalogProcess.Attributes): Seq[UUID] =
+    attributes.certified.flatMap(extractIdsFromAttribute) ++
+      attributes.declared.flatMap(extractIdsFromAttribute) ++
+      attributes.verified.flatMap(extractIdsFromAttribute)
+
+  private def extractIdsFromAttribute(attribute: CatalogProcess.Attribute): Seq[UUID] = {
+    val fromSingle: Seq[UUID] = attribute.single.toSeq.map(_.id)
+    val fromGroup: Seq[UUID]  = attribute.group.toSeq.flatMap(_.map(_.id))
+
+    fromSingle ++ fromGroup
+  }
+
+  private def convertToApiAttributes(
```

Comment:
in this repo we used the approach of implicit classes for conversions.
Probably we should use just one of them